### PR TITLE
sftp updated to allow authentication by key

### PR DESCRIPTION
**Pull request by deloittem**
This pull request is linked to the following PR:
- irma-frontend: #128 
- irma-brain: #34
- irma-probe: #69

### CODE REVIEW

File: irma/ftp/sftp.py [hunk 1]
```diff
@@ -53,7 +53,12 @@ def _connect(self):
             self._conn.window_size = pow(2, 27)
             self._conn.packetizer.REKEY_BYTES = pow(2, 32)
             self._conn.packetizer.REKEY_PACKETS = pow(2, 32)
-            self._conn.connect(username=self._user, password=self._passwd)
+            if self._auth == 'key':
+                pkey = RSAKey.from_private_key_file('/home/irma/.ssh/id_rsa')
```
Comment (ch0k0bn) [inline irma/ftp/sftp.py]: seems strange to login as frontend or probe user with a key in irma's home directory. Means also that all accounts are available with a single key.

Comment (deloittem) [inline irma/ftp/sftp.py]: what do you mean by "all accounts are available" ? To be accepted, the public key must be explicitly set in <user>/.ssh/authorized_keys

Comment (ch0k0bn) [inline irma/ftp/sftp.py]: All accounts was for "probe and frontend" the virtual users used only for ftp in irma. But got your idea of an app key.

Comment (deloittem) [inline irma/ftp/sftp.py]: Another reason: systems hardened with passwordAuth = NO in sshd_config

Comment (ch0k0bn) [inline irma/ftp/sftp.py]: good point. I want to keep sftp behaviour as close as FTP as possible. Meaning virtual users chrooted with password auth.
Maybe a config with PasswordAuthentication disabled for all users and enabled only for sftp group (that could only do sftp not ssh).
Anyway will try to find a good place for the app private key and review your PR.

Comment (ch0k0bn) [inline irma/ftp/sftp.py]: @deloittem  could you move RSA private key path to config file ? everything looks good otherwise.

**Comment by deloittem** [comment]
Private key path has been added as a parameter + check in each configuration parser that the file exists

**Comment by ch0k0bn** [comment]
merged in testing (with new param addition for FTPs)

**Pull request closed without merging**